waspc/data/Generator/templates/react-app/src/operations/resources.js

```diff
@@ -22,19 +25,21 @@ export function addResourcesUsedByQuery(queryCacheKey, resources) {
   }
 }
 
-/**
- * Invalidates all queries that are using specified resources.
- * @param {string[]} resources - Names of resources.
- */
-export async function invalidateQueriesUsing(resources) {
-  const queryClient = await queryClientInitialized
-
-  const queryCacheKeysToInvalidate = getQueriesUsingResources(resources)
-  queryCacheKeysToInvalidate.forEach(
-    queryCacheKey => queryClient.invalidateQueries(queryCacheKey)
+export function registerActionInProgress(optimisticUpdateTuples) {
+  optimisticUpdateTuples.forEach(
+    ({ queryKey, updateQueryFn}) => updateHandlers.add(queryKey, updateQueryFn)
   )
 }
 
+export async function registerActionDone(resources, optimisticUpdateTuples) {
+  optimisticUpdateTuples.forEach(({ queryKey }) => updateHandlers.remove(queryKey))
+  await invalidateQueriesUsing(resources)
+}
+
+export function getPendingUpdatesForQuery(queryKey) {
```

Comment:
Ok so we introduced term "update". Update would be an action that is affecting that query? Maybe we should kick the term update and instead use existing terms, something like "relatedAction"? Ok, maybe not hm, just thinking out loud.

Reply:
I can change it to `getPendingOptimisticUpdatesForQuery`.